decode.go

```diff
@@ -253,6 +253,16 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 
 	// Embedded message
 	case reflect.Struct:
+		bm := reflect.TypeOf((*encoding.BinaryUnmarshaler)(nil)).Elem()
```

Comment:
In fact I use it only once - why put it global? Don't you think the compiler will make this a const anyway?